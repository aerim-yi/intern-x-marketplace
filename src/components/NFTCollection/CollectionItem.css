.collection-item {
    display: flex;
    height: 50%;
    flex-direction: row;
    flex: 3;
    justify-content: space-between;
    align-items: left;
    padding: 0.5rem;
    margin: 1rem 0;
    background-color: #ba5a5a;
  }
  
  .collection-item__description {
    display: flex;
    height: 60%;
    flex-direction: row;
    gap: 1rem;
    align-items: flex-end;
    flex-flow: column-reverse;
    justify-content: flex-start;
    flex: 5;
  }
  
  .collection-item h2 {
    color: #3a3a3a;
    font-size: 1rem;
    flex: 1;
    margin: 0 1rem;
    color: white;
  }
  
  .collection-item__price {
    font-size: 1rem;
    font-weight: bold;
    color: white;
    background-color: #40005d;
    border: 1px solid white;
    padding: 0.5rem;
    border-radius: 12px;
  }
  
  @media (min-width: 100) {
    .collection-item__description {
      flex-direction: row;
      align-items: center;
      justify-content: flex-start;
      flex: 1;
    }
  
    .collection-item__description h2 {
      font-size: 1.25rem;
    }
  
    .collection-item__price {
      font-size: 1.25rem;
      padding: 0.5rem 1.5rem;
    }
  }