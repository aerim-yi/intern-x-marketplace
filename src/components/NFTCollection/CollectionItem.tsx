import React from 'react';
import Card from '../Card/Card';
import './Collection.css';
import { Button } from 'react-bootstrap'

interface Props {
  name: string | null | undefined;
  url: string;
}

const CollectionItem = (props: Props) => {
  return (
    <Card className="collections">
      <img alt="" src={props.url} data-testid="orderCardImg"/>
      <h5 data-testid="orderCardName">{props.name}</h5>
      <p>Sales Price: 10.80 ETH</p>
      <Button variant="info">Buy Now</Button>
    </Card>
  );
}

export default CollectionItem;
