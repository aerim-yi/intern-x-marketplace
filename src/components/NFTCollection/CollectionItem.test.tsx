import { render } from '@testing-library/react'
import CollectionItem from './CollectionItem'

const testProps = {
    name: 'collectionName',
    url: 'http.test.com',
    price: 'price',
    orderId: 'orderId',
}

describe('CollectionItem component', () => {
    test('match snapshot', () => {
        const { container } = render(
            <CollectionItem
                name={testProps.name}
                url={testProps.url}
                price={testProps.price}
                orderId={testProps.orderId} />
        );
        expect(container).toMatchSnapshot()
    })
})
