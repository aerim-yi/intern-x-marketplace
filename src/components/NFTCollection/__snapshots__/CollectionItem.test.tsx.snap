// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component match snapshot 1`] = `
<div>
  <div
    class="card collections"
  >
    <div
      class="cardWrapper"
    >
      <img
        alt=""
        data-testid="CollectionItem_Img"
        src="http.test.com"
      />
    </div>
    <h5
      class="cardText"
      data-testid="CollectionItem_Name"
    >
      collectionName
    </h5>
    <p>
      Sales Price: 
      price
       ETH
    </p>
    <button
      class="btn btn-info"
      type="button"
    >
      Buy Now
    </button>
  </div>
</div>
`;
